spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -397,19 +397,34 @@ class CometSparkSessionExtensions
             op,
             CometGlobalLimitExec(_, op, op.limit, op.child, SerializedPlan(None)))
 
-        case op: CollectLimitExec
-            if isCometNative(op.child) && CometConf.COMET_EXEC_COLLECT_LIMIT_ENABLED.get(conf)
-              && isCometShuffleEnabled(conf)
-              && op.offset == 0 =>
-          QueryPlanSerde
-            .operator2Proto(op)
-            .map { nativeOp =>
-              val cometOp =
-                CometCollectLimitExec(op, op.limit, op.offset, op.child)
-              CometSinkPlaceHolder(nativeOp, op, cometOp)
+        case op: CollectLimitExec =>
+          val fallbackReasons = new ListBuffer[String]()
+          if (!CometConf.COMET_EXEC_COLLECT_LIMIT_ENABLED.get(conf)) {
+            fallbackReasons += s"${CometConf.COMET_EXEC_COLLECT_LIMIT_ENABLED.key} is false"
+          }
+          if (!isCometShuffleEnabled(conf)) {
+            fallbackReasons += "Comet shuffle is not enabled"
+          }
```

Comment:
I'm actually not sure why we do not support `CollectLimitExec` if Comet shuffle is disabled :thinking: